ydb/docs/en/core/dev/example-app/python/index.md

```diff
@@ -21,217 +21,513 @@ Next, from the same working directory, run the command to start the test app. Th
+           SET air_date = CurrentUtcDate()
+           WHERE series_id = $seriesId AND season_id = $seasonId AND episode_id = $episodeId;
+           """
+
+           # Get newly created transaction id
+           tx = session.transaction(ydb.QuerySerializableReadWrite()).begin()
+
+           # Execute data query.
+           # Transaction control settings continues active transaction (tx)
+           with tx.execute(
+               query,
+               {
+                   "$seriesId": (series_id, ydb.PrimitiveType.Int64),
+                   "$seasonId": (season_id, ydb.PrimitiveType.Int64),
+                   "$episodeId": (episode_id, ydb.PrimitiveType.Int64),
+               },
+           ) as _:
+               pass
+
+           print("\n> explicit TCL call")
+
+           # Commit active transaction(tx)
+           tx.commit()
+
+       return pool.retry_operation_sync(callee)
+   ```
+
+- Asynchronous
+
+   ```python
+   async def explicit_transaction_control(
+       pool: ydb.aio.QuerySessionPoolAsync, series_id, season_id, episode_id
+   ):
+       async def callee(session: ydb.aio.QuerySessionAsync):
+           query = """
+           DECLARE $seriesId AS Int64;
+           DECLARE $seasonId AS Int64;
+           DECLARE $episodeId AS Int64;
+
+           UPDATE episodes
+           SET air_date = CurrentUtcDate()
+           WHERE series_id = $seriesId AND season_id = $seasonId AND episode_id = $episodeId;
+           """
+
+           # Get newly created transaction id
+           tx = await session.transaction(ydb.QuerySerializableReadWrite()).begin()
+
+           # Execute data query.
+           # Transaction control settings continues active transaction (tx)
+           async with await tx.execute(
+               query,
+               {
+                   "$seriesId": (series_id, ydb.PrimitiveType.Int64),
+                   "$seasonId": (season_id, ydb.PrimitiveType.Int64),
+                   "$episodeId": (episode_id, ydb.PrimitiveType.Int64),
+               },
+           ) as _:
+               pass
+
+           print("\n> explicit TCL call")
+
+           # Commit active transaction(tx)
+           await tx.commit()
+
+       return await pool.retry_operation_async(callee)
+   ```
+
+{% endlist %}
+
+However, it is worth remembering that a transaction can be opened implicitly at the first request. It could be commited automatically with the explicit indication of the `commit_tx=True` flag.
+Implicit transaction management is preferable because it requires fewer server calls. An example of implicit control will be demonstrated in the next block.
+
+## Huge Selects {#huge-selects}
+
+To perform `SELECT` operations with an unlimited number of found rows, you must also use the `transaction().execute(query)` method. As mentioned above, the result of the work is an iterator - unlike `pool.execute_with_retries(query)`, it allows you to go through the selection without first loading it into memory.
```

Comment:
```suggestion
If a `SELECT` query is expected to return a potentially large number of rows, it is recommended to use the `tx.execute` method instead of `pool.execute_with_retries` to avoid excessive memory consumption on the client side. Instead of buffering the entire result set into memory, `tx.execute` returns an iterator for each top-level `SELECT` statement in the query.
```